src/core/reactor_backend.cc

```diff
@@ -882,6 +884,18 @@ reactor_backend_epoll::wait_and_process(int timeout, const sigset_t* active_sigm
             auto op = evt.events ? EPOLL_CTL_MOD : EPOLL_CTL_DEL;
             ::epoll_ctl(_epollfd.get(), op, pfd->fd.get(), &evt);
         }
+        else if (has_error && pfd->events_requested == 0) {
```

Comment:
Yes you are right it should not be possible. I added it as a kind of a safety check to make sure we are not removing the fd from the epoll set in case we are still interested in anything. 

Given this bit of code is already a bit complicated with the different bit manipulations and their order mattering I wanted to make it resistent to any codes change that could potentially break this assumption.